packages/web/components/complex/add-conc-liquidity.tsx

```diff
@@ -0,0 +1,908 @@
+  { ssr: false }
+);
+
+export const AddConcLiquidity: FunctionComponent<
+  {
+    addLiquidityConfig: ObservableAddConcentratedLiquidityConfig;
+    actionButton: ReactNode;
+    getFiatValue?: (coin: CoinPretty) => PricePretty | undefined;
+    onRequestClose: () => void;
+  } & CustomClasses
+> = observer(
+  ({
+    className,
+    addLiquidityConfig,
+    actionButton,
+    getFiatValue,
+    onRequestClose,
+  }) => {
+    const router = useRouter();
+    const { id: poolId } = router.query as { id: string };
+    const { derivedDataStore } = useStore();
+
+    // initialize pool data stores once root pool store is loaded
+    const { poolDetail, superfluidPoolDetail } =
+      typeof poolId === "string"
+        ? derivedDataStore.getForPool(poolId as string)
+        : {
+            poolDetail: undefined,
+            superfluidPoolDetail: undefined,
+          };
+    const pool = poolDetail?.pool;
+
+    // user analytics
+    const { poolName } = useMemo(
+      () => ({
+        poolName: pool?.poolAssets
+          .map(({ amount }) => amount.denom)
+          .join(" / "),
+        poolWeight: pool?.weightedPoolInfo?.assets
+          .map((poolAsset) => poolAsset.weightFraction.toString())
+          .join(" / "),
+      }),
+      [pool?.poolAssets, pool?.weightedPoolInfo?.assets]
+    );
+
+    return (
+      <div className={classNames("flex flex-col gap-8", className)}>
+        {(() => {
+          switch (addLiquidityConfig.modalView) {
+            case "overview":
+              return (
+                <Overview
+                  pool={pool}
+                  poolName={poolName}
+                  poolDetail={poolDetail}
+                  superfluidPoolDetail={superfluidPoolDetail}
+                  addLiquidityConfig={addLiquidityConfig}
+                  onRequestClose={onRequestClose}
+                />
+              );
+            case "add_manual":
+              return (
+                <AddConcLiqView
+                  getFiatValue={getFiatValue}
+                  pool={pool}
+                  addLiquidityConfig={addLiquidityConfig}
+                  actionButton={actionButton}
+                />
+              );
+            case "add_managed":
+              return null;
+          }
+        })()}
+      </div>
+    );
+  }
+);
+
+const Overview: FunctionComponent<
+  {
+    pool?: ObservableQueryPool;
+    poolName?: string;
+    poolDetail?: ObservablePoolDetail;
+    superfluidPoolDetail?: ObservableSuperfluidPoolDetail;
+    addLiquidityConfig: ObservableAddConcentratedLiquidityConfig;
+    onRequestClose: () => void;
+  } & CustomClasses
+> = observer(
+  ({
+    addLiquidityConfig,
+    pool,
+    poolName,
+    superfluidPoolDetail,
+    poolDetail,
+    onRequestClose,
+  }) => {
+    const { priceStore, queriesExternalStore } = useStore();
+    const router = useRouter();
+    const { id: poolId } = router.query as { id: string };
```

Comment:
IMO we should get the poolId from the `addLiquidityConfig`, as we may not have a query route to rely on in some use cases. Please update this everywhere the router.query is used.